File: program/find-lcm-of-two-numbers-by-recursion/find_lcm_of_two_numbers_by_recursion.php
<?php 

function gcd($x, $y)
{
    return $x === 0 ? $y : gcd($y % $x, $x);
}

function lcm($x, $y)
{
    return ($x / gcd($x, $y)) * $y;
}

echo(lcm(12, 18));

?>

File: program/program/find-lcm-of-two-numbers-by-recursion/find_lcm_of_two_numbers_by_recursion.pl
#!/usr/bin/perl -w
use strict;
 
sub gcd {
  my ($m, $n) = @_;
  if ($n == 0) {
    return $m;
  }
 
  return gcd($n, $m % $n);
}

sub lcm {
  my ($x, $y) = @_;
  return ($x / gcd($x, $y)) * $y;
}

print lcm(12, 18);
